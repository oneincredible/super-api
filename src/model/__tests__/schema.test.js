const { createSchema } = require('../schema');
const { storages } = require('../../util/model');

describe('Schema generation', () => {
  Object.values(storages).forEach(StorageAdapter => {
    describe(StorageAdapter.getName(), () => {
      it('generates expected schema', () => {
        const statements = createSchema(StorageAdapter);
        statements.forEach(statement => {
          expect(statement).toMatchSnapshot();
        });
      });
    });
  });
});
