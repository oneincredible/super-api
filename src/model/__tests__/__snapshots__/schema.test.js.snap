// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema generation bike generates expected schema 1`] = `
"CREATE TABLE \\"bike_revision\\" (
  id uuid NOT NULL,
  revision BIGSERIAL,
  \\"price_id\\" uuid REFERENCES \\"price\\" (id),
  \\"brand\\" text,
  \\"wheel_size\\" integer,
  \\"delivery_date\\" timestamptz,
  UNIQUE(id, revision)
);"
`;

exports[`Schema generation bike generates expected schema 2`] = `
"CREATE TABLE \\"bike\\" (
  id uuid NOT NULL,
  revision BIGINT NOT NULL,
  PRIMARY KEY (id),
  FOREIGN KEY (id, revision) REFERENCES \\"bike_revision\\" (id, revision)
);"
`;

exports[`Schema generation bike generates expected schema 3`] = `
"CREATE TABLE \\"bike_wheels\\" (
  bike_id uuid NOT NULL REFERENCES \\"bike\\" (id),
  wheel_id uuid NOT NULL REFERENCES \\"wheel\\" (id),
  PRIMARY KEY (bike_id, wheel_id)
);"
`;

exports[`Schema generation price generates expected schema 1`] = `
"CREATE TABLE \\"price_revision\\" (
  id uuid NOT NULL,
  revision BIGSERIAL,
  \\"amount\\" float,
  \\"currency\\" text,
  UNIQUE(id, revision)
);"
`;

exports[`Schema generation price generates expected schema 2`] = `
"CREATE TABLE \\"price\\" (
  id uuid NOT NULL,
  revision BIGINT NOT NULL,
  PRIMARY KEY (id),
  FOREIGN KEY (id, revision) REFERENCES \\"price_revision\\" (id, revision)
);"
`;

exports[`Schema generation wheel generates expected schema 1`] = `
"CREATE TABLE \\"wheel_revision\\" (
  id uuid NOT NULL,
  revision BIGSERIAL,
  \\"size\\" float,
  \\"thickness\\" float,
  UNIQUE(id, revision)
);"
`;

exports[`Schema generation wheel generates expected schema 2`] = `
"CREATE TABLE \\"wheel\\" (
  id uuid NOT NULL,
  revision BIGINT NOT NULL,
  PRIMARY KEY (id),
  FOREIGN KEY (id, revision) REFERENCES \\"wheel_revision\\" (id, revision)
);"
`;
